hal/network/ncp_client/sara/sara_ncp_client.cpp

```diff
@@ -938,6 +940,30 @@ int SaraNcpClient::waitReady(bool powerOn) {
     return SYSTEM_ERROR_NONE;
 }
 
+int SaraNcpClient::checkImsiForNetConf() {
+    int imsiCount = 0;
+    char buf[32] = {};
+    do {
+        auto resp = parser_.sendCommand(UBLOX_CIMI_TIMEOUT, "AT+CIMI");
+        memset(buf, 0, sizeof(buf));
+        int imsiLength = 0;
+        if (resp.hasNextLine()) {
+            imsiLength = CHECK(resp.readLine(buf, sizeof(buf)));
```

Comment:
Might want to do `sizeof(buf) - 1` to avoid potential to pass a non-null terminated string to `strlen(buf)` below, or better pass `imsiLength` to `networkConfigForImsi(buf, imsiLength)` directly.

Reply:
:+1: Passing `imsiLength` sounds better.